dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManager.kt

```diff
@@ -57,14 +58,63 @@ class SingleDataRequestManager(
         val storedDataRequests = storeDataRequestsAndAddThemToListForEachReportingPeriodIfNotAlreadyExisting(
             singleDataRequest, datalandCompanyId,
         )
-        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+        sendSingleDataRequestEmailMessages(
             userAuthentication = DatalandAuthentication.fromContext() as DatalandJwtAuthentication,
             singleDataRequest = singleDataRequest,
             datalandCompanyId,
         )
         return storedDataRequests
     }
 
+    private fun sendSingleDataRequestEmailMessages(
+        userAuthentication: DatalandJwtAuthentication,
+        singleDataRequest: SingleDataRequest,
+        datalandCompanyId: String,
+    ) {
+        if (singleDataRequest.reportingPeriods.isEmpty()) return
```

Comment:
Das ist eigentlich eher ein validty check des singe-data-requests selbst und sollte oben in der processer-Funktion ganz am Anfang überprüft werden.
Ähnlich wie im `BulkDataRequestManager` => dort wird ganz am Anfang der processer-Funktion die Funktion `assureValidityOfRequests` aufgerufen.  Die überprüft erstmal ob alles so passt wie es ist, bevor mit der weiteren Prozessierung fortgefahren wird.

Reply:
Da stimme ich zu. Aber das änder glaube ich Benedikt schon in seinem Branch.

Reply:
Dann würde ich Benedikts fix cherry-picken.

So würde ich das sehr ungern nach main lassen.

Reply:
Hab eine Lösung implementiert